src/data/update.ts

```diff
@@ -26,8 +26,15 @@ export const update = (api: VectorOperationsApi, namespace: string) => {
   return async (options: UpdateVectorOptions): Promise<void> => {
     validator(options);
 
+    const requestOptions = {
+      id: options['id'],
+      values: options['values'],
+      sparseValues: options['sparseValues'],
+      setMetadata: options['metadata'],
+    };
```

Comment:
**praise:** I like this for re-mapping things. 👍

Reply:
thanks! me too! but i'm going to revise again so as to not break the principle Jen mentions [here](https://github.com/pinecone-io/pinecone-ts-client/pull/83#discussion_r1303365302)

Reply:
This reads nicely, but I think it means we lose the flexibility of being able to update any properties that get added in the future.

Reply:
Got it, catching up on the follow-up comments.

Reply:
one more follow-up -- this _is_ okay for now because our generated API code doesn't allow the forward flexibility we thought it did, so we'll keep this implementation.